### PR TITLE
feat(api): #MC-161 add trace to api

### PR DESCRIPTION
**Pull request by AyaNADINI**
## Describe your changes
Providing a way to determine what exactly is happening with HTTP request, to diagnose unexpected or undesirable behavior.
To deal with this, we add API trace to trace HTTP request on specific routes.
Trace annotation contains two parameters :
value (String): Set action name
body (Boolean): Set if the request body should be saved(by default body takes true)

## Checklist tests
Create agenda in "http://localdev:8090/calendar#/"
check in mongodb using `db.getCollection('traces').find({"action":"CREATE_CALENDAR"})`

## Issue ticket number and link
MC-161 : https://entsupport.gdapublic.fr/browse/MC-161

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x]  I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 1]
```diff
@@ -149,6 +153,7 @@ public void shareCalendar(final HttpServerRequest request) {
     @Put("/share/json/:id")
     @ApiDoc("Share calendar by id.")
     @SecuredAction(value = "calendar.manager", type = ActionType.RESOURCE)
+    @Trace(value="SHARE CALENDAR_SUBMIT")
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:156]: utilise toujours un _, pas d'espace ==> SHARE_CALENDAR_SUBMIT

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 2]
```diff
@@ -177,6 +182,7 @@ public void shareCalendarSubmit(final HttpServerRequest request) {
     @Put("/share/remove/:id")
     @ApiDoc("Remove calendar by id.")
     @SecuredAction(value = "calendar.manager", type = ActionType.RESOURCE)
+    @Trace(value="SHARE CALENDAR_REMOVE")
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:185]: utilise toujours un _ (pas d'espace) ==> SHARE_CALENDAR_REMOVE

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 3]
```diff
@@ -93,11 +94,12 @@ public void view(HttpServerRequest request) {
     @Get("/calendars")
     @SecuredAction("calendar.view")
     public void listCalendars(HttpServerRequest request) {
-        list(request);
+
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:97]: pourquoi supprimer le list ?

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 4]
```diff
@@ -185,6 +191,7 @@ public void removeShareCalendar(final HttpServerRequest request) {
     @ApiDoc("Share calendar by id.")
     @ResourceFilter(ShareEventConf.class)
     @SecuredAction(value = "calendar.manager", type = ActionType.RESOURCE)
+    @Trace(value="SHARE_RESOURCE")
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:194]: SHARE_CALENDAR

File: src/main/java/net/atos/entng/calendar/controllers/EventController.java [hunk 5]
```diff
@@ -109,6 +113,7 @@ public void shareEvent(final HttpServerRequest request) {
     @ApiDoc("Share calendar by id.")
     @ResourceFilter(ShareEventConf.class)
     @SecuredAction(value = "calendar.manager", type = ActionType.RESOURCE)
+    @Trace(value="SHARE_RESOURCE")
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/controllers/EventController.java:116]: SHARE_EVENT

**Review by alicedraillard — APPROVED** [review]

**Review by vava56754 — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 6]
```diff
@@ -98,6 +99,7 @@ public void listCalendars(HttpServerRequest request) {
 
     @Post("/calendars")
     @SecuredAction("calendar.create")
+    @Trace(value="CREATE_CALENDAR")
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:102]: Est-ce que ce ne serait pas mieux de faire une liste regroupant tous les traces dans un fichier appart. En peu comme Field.java

Comment (JinDouang) [reply src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:102]: Je suis d'accord, une constant "Actions" pour qu'il soit spécifique aux traces

Comment (JinDouang) [reply src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:102]: `@Trace(Actions.CREATE_CALENDAR` pas besoin de rajouter "value"

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/java/net/atos/entng/calendar/core/constants/Actions.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    public static final String UPDATE_CALENDAR = "UpdateCalendar";
+    public static final String DELETE_CALENDAR = "DeleteCalendar";
+    public static final String SHARE_CALENDAR_SUBMIT = "ShareCalendarSubmit";
+    public static final String SHARE_CALENDAR_REMOVE = "ShareCalendarRemove";
+    public static final String SHARE_CALENDAR = "ShareCalendar";
+    public static final String CREATE_EVENT = "CreateEvent";
+    public static final String UPDATE_EVENT = "UpdateEvent";
+    public static final String DELETE_EVENT = "DeleteEvent";
+    public static final String SHARE_EVENT = "ShareEvent";
+    public static final String IMPORT_ICL = "ImportIcal";
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/core/constants/Actions.java:20]: IMPORT_ICAL

**Review by alicedraillard — APPROVED** [review]

**Comment by JinDouang** [comment]
@alicedraillard Pour les traces on ne souhaite pas stocker le payload également ?

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/core/constants/Actions.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package net.atos.entng.calendar.core.constants;
+
+public class Actions {
+
+    private Actions() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    //Trace actions
+    public static final String CREATE_CALENDAR = "createCalendar";
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/calendar/core/constants/Actions.java:10]: pourquoi pas en SNAKE_CASE ?

**Review by adrienFro — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**